docker/Dockerfile.devito

```diff
@@ -4,14 +4,38 @@
 
 # Base image with compilers
 ARG base=devitocodes/bases:cpu-gcc
+ARG petsc=""
+
+##################  Install PETSc ############################################
+FROM $base as petsc_builder 
```

Comment:
Cleaner way is

``` 

FROM builder as petsbuilder

do the petsc install


FROM $base as user

ARG PETSC=""

...

COPY --from=builder${petsc}

```


This will only build the petsc if needed by the `copy` command and you won't need the IF